include/ygm/container/oversized_bag.hpp

```diff
@@ -0,0 +1,335 @@
+
+/** @todo ----------------------------------------------unchanged----------------------------------------------*/
+template <typename Item>
+class oversized_bag : public detail::base_async_insert_value<oversized_bag<Item>, std::tuple<Item>>,
+            public detail::base_count<oversized_bag<Item>, std::tuple<Item>>,
+            public detail::base_misc<oversized_bag<Item>, std::tuple<Item>>,
+            public detail::base_iteration_value<oversized_bag<Item>, std::tuple<Item>> {
+  friend class detail::base_misc<oversized_bag<Item>, std::tuple<Item>>;
+
+ public:
+  using self_type      = oversized_bag<Item>;
+  using value_type     = Item;
+  using size_type      = size_t;
+  using for_all_args   = std::tuple<Item>;
+  using container_type = ygm::container::bag_tag;
+  
+  oversized_bag(ygm::comm &comm) : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag(ygm::comm &comm, std::initializer_list<Item> l)
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+    if (m_comm.rank0()) {
+      for (const Item &i : l) {
+        async_insert(i);
+      }
+    }
+    m_comm.barrier();
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /**
+   * @brief Construct a new oversized bag object from a directory of files, arguments ect could be different
+   */
+  oversized_bag(ygm::comm &comm, std::string dir)
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+    /**
+     * @todo Implement this
+     */
+  }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  template <typename STLContainer>
+  oversized_bag(ygm::comm          &comm,
+      const STLContainer &cont) requires detail::STLContainer<STLContainer> &&
+      std::convertible_to<typename STLContainer::value_type, Item>
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+
+    for (const Item &i : cont) {
+      this->async_insert(i);
+    }
+    m_comm.barrier();
+  }
+
+  template <typename YGMContainer>
+  oversized_bag(ygm::comm          &comm,
+      const YGMContainer &yc) requires detail::HasForAll<YGMContainer> &&
+      detail::SingleItemTuple<typename YGMContainer::for_all_args>
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+
+    yc.for_all([this](const Item &value) { this->async_insert(value); });
+
+    m_comm.barrier();
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /**
+   * @todo we'll need to close all of the files we opened
+   */
+  ~oversized_bag() { m_comm.barrier(); }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  oversized_bag(const self_type &other)  // If I remove const it compiles
```

Comment:
Is this going to be a problem or will changing whether the argument is constant or not affect our progress? If so, what are some possible solutions for this?

Reply:
I'm not 100% sure, I think we just leave it as it is in ygm::bag. This is either the move or copy constructor which there's also a chance there's not real semantics for  so it becomes `oversized_bag(const self_type &other) = delete;`